src/clj/backtype/storm/daemon/supervisor.clj

```
@@ -317,7 +317,14 @@
         (when-not (assigned-storm-ids storm-id)
           (log-message "Removing code for storm id "
                        storm-id)
-          (rmr (supervisor-stormdist-root conf storm-id))
+          (try
+            (rmr (supervisor-stormdist-root conf storm-id))
+            (catch java.io.IOException e
+              ;; Handles a race condition resulting from topology kill where this sync call
```

There's legitimate cases where an IOException is thrown here and the rmr did not go through. This is a case where we should check if we're running in Windows and take different behavior as opposed to Mac/Linux (which should rethrow the exception)